### PR TITLE
[RHCLOUD-36892] Forbid inconsistent user subscriptions

### PR DESCRIPTION
**Pull request by g-duval**
This PR adds consistency checks on email and drawer subscriptions.
It forbid user subscription it the linked event type don't have a template for the selected subscription type.

### CODE REVIEW

File: backend/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java [hunk 1]
```diff
@@ -142,7 +143,29 @@ public InstantEmailTemplate findInstantEmailTemplateByEventType(UUID eventTypeId
 
             return instantEmailTemplate;
         } catch (NoResultException e) {
-            throw new NotFoundException("Instant email template not found");
+            throw new NotFoundException(String.format("Event type '%s' doesn't support instant email subscription", eventTypeId));
+        }
+    }
+
+    public AggregationEmailTemplate findAggregationEmailTemplatesByEventType(UUID eventTypeId) {
```
Comment (gwenneg) [inline backend/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java:150]: The returned object is never used. This method could return a boolean and `FETCH` could be avoided to lower the pressure on the DB. Unless you have a plan to use the method from elsewhere and use the returned data in the near future.

File: backend/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java [hunk 2]
```diff
@@ -142,7 +143,29 @@ public InstantEmailTemplate findInstantEmailTemplateByEventType(UUID eventTypeId
...
+        try {
+            return entityManager.createQuery(hql, AggregationEmailTemplate.class)
+                .setParameter("eventTypeId", eventTypeId)
+                .getSingleResult();
+        } catch (NoResultException e) {
+            throw new NotFoundException(String.format("Event type '%s' doesn't support daily email subscription", eventTypeId));
+        }
+    }
+
+    public IntegrationTemplate findDrawerTemplateByEventType(UUID eventTypeId) {
```
Comment (gwenneg) [inline backend/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java:161]: Same comment about the pressure on the DB.

File: backend/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java (hunk 1 above)
Comment (g-duval) [reply backend/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java]: sure, I refactored those methods to not fetch results anymore

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**